go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -559,52 +559,81 @@ func (b *builtinFromUnixtime) eval(env *ExpressionEnv) (eval, error) {
 
 	switch ts := ts.(type) {
 	case *evalInt64:
+		if ts.i < 0 || ts.i >= maxUnixtime {
+			return nil, nil
+		}
```

Comment:
It's repetitive, but we have to check up front here. Some fun things:

#### amd64

Large float turns into negative int64.

```
int64(1.2345678912345678e300)
-9223372036854775808
```

#### arm64

Large float64 turns into max int64. 

```
int64(1.2345678912345678e+300)

9223372036854775807
```

This means we can't convert before checking overflow / underflow since it won't be consistent.